float64 distance
float64 angle
float64 angleX
float64 angleY
float64 angleZ
float64 accX
float64 accY
float64 accZ
